### PR TITLE
Runtime Observable Settings

### PR DESCRIPTION
**Pull request by psuzn**
This adds support for the wrapper around the normal setting (#155 ) to make it runtime observable.  

@russhwolf  I have named it `runtime-observable` but we can rename it if there is a better name. 

I also have added the `checkObservable` flag on the converter extension function, so one can just use the `no-arg` initializer and use the converter to make it observable.  In some platforms, the instance returned by `no-arg` is already observable; in that case, the converter doesn't add any wrapper.

### CODE REVIEW

**Comment by russhwolf** [comment]
Thanks! I'm traveling right now so I probably won't get a chance to review this closely for a week or so.

I'm still not sure how I feel on the naming, but will probably merge it with what you have and change it before I put out a release with it.

Speaking of the next release, I already have a 1.2 branch with some other pending updates. Do you think you can rebase against that branch? If not, I can do it when I'm back. I'm guessing the only major change needed will be the gradle files because I converted the old buildsrc code to use script plugins.

**Comment by russhwolf** [comment]
To fix the CI failure, run the `apiDump` gradle task from the new module and push the api files it generates to this branch.

**Comment by psuzn** [comment]
@russhwolf, any update on this?

File: multiplatform-settings-runtime-observable/src/commonMain/kotlin/Converter.kt [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ *  Creates a [RuntimeObservableSettingsWrapper] wrapper around the [Settings] instance.
+ *
+ *  @param checkObservable flag that indicates whether to skip the observable check for current
+ *  instance or not. If it is false and the [Settings] instance is already a [ObservableSettings]
+ *  it doesn't create a [RuntimeObservableSettingsWrapper] for it.
+ *
+ */
+public fun Settings.toRuntimeObservable(checkObservable: Boolean = true): ObservableSettings =
+    if (checkObservable && this is ObservableSettings) this
+    else RuntimeObservableSettingsWrapper(this)
```
Comment (russhwolf) [inline multiplatform-settings-runtime-observable/src/commonMain/kotlin/Converter.kt:32]: Is there a use-case for `checkObservable = false`?

File: multiplatform-settings-runtime-observable/src/commonTest/kotlin/RuntimeObservableSettingsWrapperTest.kt [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+internal class RuntimeObservableSettingFactory : Settings.Factory {
+
+    override fun create(name: String?): Settings {
+        return RuntimeObservableSettingsWrapper(MapSettings())
+    }
+}
+
+
+class RuntimeObservableSettingsWrapperTest : BaseSettingsTest(
```
Comment (russhwolf) [inline multiplatform-settings-runtime-observable/src/commonTest/kotlin/RuntimeObservableSettingsWrapperTest.kt:36]: Does this class test anything that couldn't be checked via `SettingsExtensionTest`? I'm inclined to combine them and just have the test interact with public API (and then `RuntimeObservableSettingsWrapper` can move from `internal` to `private`)

**Review by russhwolf — COMMENTED** [review]
Sorry to leave this sitting. I've had less time over the last couple weeks than I'd hoped. I plan to merge this soon and probably make a couple tweaks on top of it, but had a couple clarifying questions first.

File: multiplatform-settings-runtime-observable/src/commonMain/kotlin/Converter.kt (hunk 1 above)
Comment (psuzn) [reply multiplatform-settings-runtime-observable/src/commonMain/kotlin/Converter.kt:32]: My thought was to give more control to the end user but now I think about it, there are not many use cases for it.

File: multiplatform-settings-runtime-observable/src/commonTest/kotlin/RuntimeObservableSettingsWrapperTest.kt (hunk 2 above)
Comment (psuzn) [reply multiplatform-settings-runtime-observable/src/commonTest/kotlin/RuntimeObservableSettingsWrapperTest.kt:36]: There is only the `delegateTest` on the `RuntimeObservableSettingsWrapper` which checks if the wrapper and delegate are consistent. I can move it to `SettingsExtensionTest` but we'll still need access to `RuntimeObservableSettingsWrapper` for testing.

**Pull request merged**